src/hooks/useInfiniteScroll.ts

```diff
@@ -12,17 +12,18 @@ type FetchMoreOpts = {
  * fetchMore should fetch more results and update any state tied to your UI
  * returns how many items it fetched and if there is a next page
  */
-export type FetchMore = (_opts: FetchMoreOpts) => Promise<{
+export type FetchMore<T> = (_opts: FetchMoreOpts) => Promise<{
   itemsFetched: number;
   hasNextPage: boolean;
+  data: T | null;
```

Comment:
Return the fetched data to allow consuming UI to react to these updates. Previously `fetchMore` itself needed to include the UI updater, but that made for unnecessarily tight coupling